File: Examples/CSharp/AsposePdfFacades/Working-Document/SetViewerPreference.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace CSharp.AsposePDFFacades.WorkingDocument
{
    public class SetViewerPreference
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_WorkingDocuments();
            //open document
            PdfContentEditor contentEditor = new PdfContentEditor();
            contentEditor.BindPdf(dataDir+ "SetViewerPreference.pdf");

            //change Viewer Preferences
            contentEditor.ChangeViewerPreference(ViewerPreference.CenterWindow);
            contentEditor.ChangeViewerPreference(ViewerPreference.HideMenubar);
            contentEditor.ChangeViewerPreference(ViewerPreference.PageModeUseNone);

            //save output PDF file
            contentEditor.Save(dataDir+ "SetViewerPreference_out.pdf");
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Working-Document/SetViewerPreference.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.WorkingDocument
    Public Class SetViewerPreference
        Public Shared Sub Run()
            ' ExStart:SetViewerPreference
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_WorkingDocuments()
            ' Open document
            Dim contentEditor As New PdfContentEditor()
            contentEditor.BindPdf(dataDir & "SetViewerPreference.pdf")

            ' Change Viewer Preferences
            contentEditor.ChangeViewerPreference(ViewerPreference.CenterWindow)
            contentEditor.ChangeViewerPreference(ViewerPreference.HideMenubar)
            contentEditor.ChangeViewerPreference(ViewerPreference.PageModeUseNone)

            ' Save output PDF file
            contentEditor.Save(dataDir & "SetViewerPreference_out_.pdf")
            ' ExEnd:SetViewerPreference
        End Sub
    End Class
End Namespace